src/utils/utils_part.h

```diff
@@ -50,6 +51,39 @@ static inline ocf_part_id_t ocf_part_class2id(ocf_cache_t cache, uint64_t class)
 	return PARTITION_DEFAULT;
 }
 
+static inline uint32_t ocf_part_get_occupancy(ocf_cache_t cache,
+		ocf_part_id_t part_id)
+{
+	return cache->user_parts[part_id].runtime->curr_size;
+}
+
+static inline uint32_t ocf_part_get_min_size(ocf_cache_t cache,
+		ocf_part_id_t part_id)
+{
+	uint64_t ioclass_size;
+
+	ioclass_size = cache->user_parts[part_id].config->min_size *
+		cache->conf_meta->cachelines;
+
+	ioclass_size /= 100;
+
+	return (uint32_t)ioclass_size;
+}
+
+
+static inline uint32_t ocf_part_get_max_size(ocf_cache_t cache,
+		ocf_part_id_t part_id)
+{
+	uint64_t ioclass_size;
+
+	ioclass_size = cache->user_parts[part_id].config->max_size *
+		cache->conf_meta->cachelines;
```

Comment:
aren both operands 32-bit? If so, there is a potential for overflow here despite assigning to wider type.

Reply:
please make sure max_size = 100 allows you to fill entire cache with cache device large enough to contain more than MAX_UINT32 / 100. This should be easy on null block device

Reply:
I did the test, and it works fine